### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by WatanabeToshimitsu**
## What?
- [x] Separate logic and view
- [x] Use react-hook-form in some form
- [x] Share reusable component
- [x] Create new component to split some too big component
- [x] Fix usage of SWR 

## Why?
- 実装を優先してぐちゃっとした部分が多いから
- SWR の使い方が根本的に間違っていたから

## See also [Optional]
- https://www.notion.so/humandatawarelab/Board-3497e916acdb4b39b584981983789bbc?p=f1e0fceafab24c4386cf4bb104b0278c

### CODE REVIEW

**Review by d-hayashi — APPROVED** [review]
LGTM!

**Comment by hdl-service** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/dataware-tools/app-data-browser-next/pull/61#pullrequestreview-700516457" title="Approved">d-hayashi</a>*, *<a href="https://github.com/dataware-tools/app-data-browser-next/pull/61#" title="Author self-approved">WatanabeToshimitsu</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=dataware-tools%2Fapp-data-browser-next).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/dataware-tools/app-data-browser-next/blob/master/OWNERS)~~ [WatanabeToshimitsu,d-hayashi]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
